### PR TITLE
feat(oidc): allow returning of parent errors to client

### PR DESCRIPTION
**Pull request by muhlemmer**
# Which Problems Are Solved

Currently the OIDC API of ZITADEL only prints parent errors to the logs. Where 4xx status are typically warn level and 5xx error level. This makes it hard to debug certain errors for client in multi-instance environments like ZITADEL cloud, where there is no direct access to logs. In case of support requests we often can't correlate past log-lines to the error that was reported.

This change adds the possibility to return the parent error in the response to the OIDC client. For the moment this only applies to JSON body responses, not error redirects to the RP.

# How the Problems Are Solved

- New instance-level feature flag: `debug_oidc_parent_error`
- Use the new `WithReturnParentToClient()` function from the oidc lib introduced in https://github.com/zitadel/oidc/pull/629 for all cases where `WithParent` was already used and the request context is available.

# Additional Changes

none

# Additional Context

- Depends on: https://github.com/zitadel/oidc/pull/629
- Related to: https://github.com/zitadel/zitadel/issues/8362

### CODE REVIEW

**Review by livio-a — COMMENTED** [review]
Unsure if allowing this is a problem, because of the potential exposure of sensitive information or if we should more focus of giving admins access to these information, e.g. zitadel.cloud admins being able to access to their logs

File: internal/api/grpc/feature/v2/converter.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ func instanceFeaturesToCommand(req *feature_pb.SetInstanceFeaturesRequest) *comm
 		TokenExchange:                   req.OidcTokenExchange,
 		Actions:                         req.Actions,
 		ImprovedPerformance:             improvedPerformanceListToDomain(req.ImprovedPerformance),
+		DebugOIDCParentError:            req.DebugOidcParentError,
```
[the remark this answers is not in the document]
Comment (muhlemmer) [reply internal/api/grpc/feature/v2/converter.go:45]: Done

**Comment by livio-a** [comment]
@muhlemmer looks like some test is failing for the feature flag. did not check if it's because of eventual consistency or if there something missing in the implementation

**Comment by muhlemmer** [comment]
@livio-a  should be allright now

**Comment by muhlemmer** [comment]
Please don't merge before https://github.com/zitadel/zitadel/pull/8262. Probably need to handle merge conflicts after.

**Comment by muhlemmer** [comment]
@livio-a merge conflicts are handled and should be ready now.

**Review by livio-a — APPROVED** [review]

**Pull request merged**